lib/ain-ocean/src/data_acces/masternode.rs

```diff
@@ -1,19 +1,49 @@
-use anyhow::{anyhow, Result};
-use serde::{Deserialize, Serialize};
-use serde_json;
-
 use crate::{
-    database::db_manager::{ColumnFamilyOperations, RocksDB},
+    database::db_manager::{ColumnFamilyOperations, RocksDB, SortOrder},
     model::masternode::Masternode,
 };
+use anyhow::{anyhow, Result};
+use rocksdb::IteratorMode;
+use serde_json;
 
 pub struct MasterNodeDB {
     pub db: RocksDB,
 }
 
 impl MasterNodeDB {
-    pub async fn query(&self, limit: i32, lt: i32) -> Result<Vec<Masternode>> {
-        todo!()
+    pub async fn query(
+        &self,
+        limit: i32,
+        lt: i32,
+        sort_order: SortOrder,
+    ) -> Result<Vec<Masternode>> {
+        let iterator = self.db.iterator("masternode", IteratorMode::End)?;
+        let mut master_node: Vec<Masternode> = Vec::new();
+        let collected_blocks: Vec<_> = iterator.collect();
+
+        for result in collected_blocks.into_iter().rev() {
+            let (key, value) = match result {
+                Ok((key, value)) => (key, value),
+                Err(err) => return Err(anyhow!("Error during iteration: {}", err)),
+            };
+
+            let master_stats: Masternode = serde_json::from_slice(&value)?;
+            master_node.push(master_stats);
+
+            if master_node.len() == limit as usize {
+                break;
+            }
+        }
+
+        // Sort blocks based on the specified sort order
+        match sort_order {
+            SortOrder::Ascending => master_node.sort_by(|a, b| a.block.height.cmp(&b.block.height)),
+            SortOrder::Descending => {
+                master_node.sort_by(|a, b| b.block.height.cmp(&a.block.height))
+            }
+        }
```

Comment:
We would need a From impl from `SortOrder` to RocksDB Iterator mode. 
In both SortOrder cases, we iterate from the end, get the batch and sort it. We should be iterating from start for `SortOrder::Ascending`.